scripts/app.ts

```diff
@@ -120,12 +119,33 @@ function updateIterationPath(childIdsToMove: number[], iterationPath: string): I
     return Promise.all(promises);
 }
 
-function createWorkItem(workItem: TFS_Wit_Contracts.WorkItem, copyTags: boolean, title?: string, iterationPath?: string): IPromise<TFS_Wit_Contracts.WorkItem> {
-    var patchDocument = [];
-    var fieldsToCopy = [CoreFields.Title, CoreFields.AssignedTo, CoreFields.IterationPath, CoreFields.AreaPath, CoreFields.Description];
-    if (copyTags){
+async function createWorkItem(workItem: TFS_Wit_Contracts.WorkItem, copyTags: boolean, title?: string, iterationPath?: string): Promise<TFS_Wit_Contracts.WorkItem> {
+    const context = VSS.getWebContext();
+    let patchDocument = [];
+    const currentWorkItemType = workItem.fields[CoreFields.WorkItemType];
+
+    /* Hello custom extension author - Add your custom field ref name here!*/
+    var fieldsToCopy = [CoreFields.Title, CoreFields.AssignedTo, CoreFields.IterationPath, CoreFields.AreaPath, CoreFields.Description,
+        AdditionalFields.AcceptanceCriteria, AdditionalFields.ReproSteps, AdditionalFields.SystemInfo];
+
+    // Copy any fields that are required for this work item 
+    const workItemTypeInfo = await TFS_Wit_Client.getClient().getWorkItemType(context.project.name, currentWorkItemType);
+    workItemTypeInfo.fields.forEach(f => {        
+        // Don't include iteration related fields or state, we don't want that copied from the current work item        
+        const isIgnoredField = f.referenceName === AdditionalFields.IterationId || f.referenceName === CoreFields.State; 
+        const isRequiredField = f.alwaysRequired;
+        if (isRequiredField && !isIgnoredField ){
+            if (fieldsToCopy.indexOf(f.referenceName) === -1) {
```

Comment:
field reference names should use case insensitive compare.

Reply:
Thanks!